tensorflow-ops/tests/MatrixTest.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedLists #-}
+
+import Data.Int (Int32, Int64)
+
+import Control.Monad.IO.Class (liftIO)
+import Control.Monad (replicateM_, zipWithM)
+
+import TensorFlow.GenOps.Core (square)
+import qualified TensorFlow.Core as TF
+import qualified TensorFlow.Gradient as TF
+import qualified TensorFlow.Ops as TF
+import qualified Data.Vector as V
+
+import Test.Framework (Test)
+import Test.Framework.Providers.HUnit (testCase)
+import TensorFlow.Test (assertAllClose)
+import Google.Test (googleTest)
+
+
+randomParam (TF.Shape shape) =
+  (`TF.mul` (TF.scalar 1.0)) <$> TF.truncatedNormal (TF.vector shape)
+
+reduceMean :: TF.Tensor TF.Build Float -> TF.Tensor TF.Build Float
+reduceMean xs = TF.mean xs (TF.scalar (0 :: Int32))
```

Comment:
Here is a better implementation so you don't have to do `reduceMean . reduceMean`:

```haskell
reduceMean :: TF.Tensor TF.Value Float -> TF.Tensor TF.Value Float
reduceMean xs = TF.mean xs (TF.range 0 (TF.rank xs) 1)
```

Maybe add it to `TensorFlow.Ops`